### ISSUE TITLE
case-insensitive completion is contra-productive and needs a way to be turned off

### ISSUE THREAD

**Issue by hohwille** [msys2/MSYS2-packages#4697]
### Description / Steps to reproduce the issue

```
$ shopt -u nocasematch nocaseglob
$ cd /tmp/
$ mkdir IDEasy
$ mkdir idea
$ cd IDEa
IDEasy/ idea/
```
NOTE: I pressed `tab` key twice after `cd IDEa` to trigger auto-completion.

### Expected behavior

Already after pressing `tab` key once, I expect to get `cd IDEa` be completed to
```
$ cd IDEasy
```

### Actual behavior

Auto-completion behaves case-insensitive
```
$ cd IDEa
IDEasy/ idea/
```

If you consider this as a feature, then please provide a (documented) way how to switch this off and get case-sensitive completion as this is contra-productive to several users. See also [this open question](https://unix.stackexchange.com/questions/750589/how-can-i-turn-off-git-bash-for-windows-case-insensitive-behavior).

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MSYS_NT-10.0-19045

### Are you willing to submit a PR?

I am an active OSS developer but only have Java skills. So sorry, I guess I am unable to create a PR in reasonable time.

**Comment by jeremyd2019** [comment]
Haven't investigated, but maybe `MSYS=glob:noignorecase` ?

**Comment by hohwille** [comment]
Thanks for the hint but `export MSYS=glob:noignorecase` does not change the behavior.

**Comment by dscho** [comment]
Globbing is a Bash/readline feature and case-sensitivity can be enforced via this command (which you're welcome to put into your `~/.bashrc`):

```sh
bind "set completion-ignore-case off"
```

The default should not be changed nilly-willy, as it affects many more users than just the ones who wish case-sensitive tab completion. Realistically, any default here will upset one group of passionate people, and it is the wise thing to do to avoid flip-flopping between defaults and upsetting _everyone_.

**Comment by jeremyd2019** [comment]
Yes thanks.  I knew cygwin globbing had a case-insensitive flag (I'm not so sure that prefixing it with `no` actually works, but the default seems to be case sensitive), but I didn't know that bash/readline re-created its own globbing with its own flag.

**Comment by dscho** [comment]
@jeremyd2019 I think the Cygwin globbing feature is only in effect when parsing command-lines passed via `CreateProcess()`. The completion is decoupled from that. In fact, you can even set that Bash/Readline flag on Linux and have case-insensitive tab completion on that platform, even though most Linux setups work with case-sensitive filesystems.

**Comment by jeremyd2019** [comment]
In POSIX there is also a function `glob` - https://linux.die.net/man/3/glob - which cygwin provides, and which something like a shell could use if it didn't provide its own implementation.  The particular globbing you reference is done by a function called `globify` (https://github.com/cygwin/cygwin/commit/14216d5442e04f23f20318b434b3f2bc4c43ca90) which calls that function.
